mobile_verifier/src/reward_shares.rs

```diff
@@ -121,10 +141,15 @@ impl TransferRewards {
                 reward: Some(proto::mobile_reward_share::Reward::GatewayReward(
                     proto::GatewayReward {
                         hotspot_key: hotspot_key.into(),
-                        dc_transfer_reward: (reward * reward_scale)
+                        dc_transfer_reward: (reward.bones * reward_scale)
                             .round_dp_with_strategy(0, RoundingStrategy::ToZero)
                             .to_u64()
-                            .unwrap_or(0),
+                            .unwrap_or_default(),
+                        rewardable_bytes: reward.bytes_rewarded,
+                        // Can someone confirm what this should be?
+                        price: (self.mobile_bone_price * dec!(1_000_000))
```

Comment:
i think you are going to need to multiply that again, to get it the mobile token price at 10^6 otherwise you will still have a value < 0 at the point of converting to a float and thus end up with a zero price value, so basically:

price: (self.mobile_bone_price * dec!(1_000_000) * dec!(1_000_000)

This will take you back to Mobile price @ 10^6 which fits with the u64 proto definition